src/main/java/org/elasticsearch/action/update/UpdateResponse.java

```diff
@@ -57,6 +58,11 @@ public UpdateResponse(ShardInfo shardInfo, String index, String type, String id,
         this.created = created;
     }
 
+    public UpdateResponse(String index, String type, String id, long version, boolean created, boolean noop) {
```

Comment:
instead of a boolean for noop, it would be better if we can have opcode, so that the client can know if it is indexed, deleted or noop